src/commonMain/kotlin/io/kotest/assertions/ktor/client/response.kt

```diff
@@ -4,83 +4,76 @@ import io.kotest.matchers.Matcher
+   this should haveHeader(HttpHeaders.CacheControl, cacheControl)
 
-infix fun HttpResponse.shouldHaveCacheControl(cacheControl: String) = this should haveCacheControl(cacheControl)
-infix fun HttpResponse.shouldNotCacheControl(cacheControl: String) = this shouldNot haveCacheControl(cacheControl)
-fun haveCacheControl(expected: String) = object : Matcher<HttpResponse> {
-   override fun test(value: HttpResponse): MatcherResult {
-      val actual = value.headers[HttpHeaders.CacheControl]
-      return MatcherResult(
-         actual == expected,
-         { "Response should have Cache-Control: $expected but had: ${actual}." },
-         { "Response should not have Cache-Control $expected." },
-      )
-   }
-}
+fun HttpResponse.shouldHaveCacheControl() = this should haveHeader(HttpHeaders.CacheControl)
+infix fun HttpResponse.shouldNotHaveCacheControl(cacheControl: String) =
+   this shouldNot haveHeader(HttpHeaders.CacheControl, cacheControl)
 
-infix fun HttpResponse.shouldHaveContentEncoding(encoding: String) = this should haveContentEncoding(encoding)
-infix fun HttpResponse.shouldNotHaveContentEncoding(encoding: String) = this shouldNot haveContentEncoding(encoding)
-fun haveContentEncoding(expected: String) = object : Matcher<HttpResponse> {
-   override fun test(value: HttpResponse): MatcherResult {
-      val actual = value.headers[HttpHeaders.ContentEncoding]
-      return MatcherResult(
-         actual == expected,
-         { "Response should have Content-Encoding: $expected but had: ${actual}." },
-         { "Response should not have Content-Encoding $expected." },
-      )
-   }
-}
+fun HttpResponse.shouldNotHaveCacheControl() = this shouldNot haveHeader(HttpHeaders.CacheControl)
+infix fun HttpResponse.shouldHaveContentEncoding(encoding: String) =
+   this should haveHeader(HttpHeaders.ContentEncoding, encoding)
+
+fun HttpResponse.shouldHaveContentEncoding() = this should haveHeader(HttpHeaders.ContentEncoding)
+
+infix fun HttpResponse.shouldNotHaveContentEncoding(encoding: String) =
+   this shouldNot haveHeader(HttpHeaders.ContentEncoding, encoding)
+
+fun HttpResponse.shouldNotHaveContentEncoding() = this shouldNot haveHeader(HttpHeaders.ContentEncoding)
+
+infix fun HttpResponse.shouldHaveHeader(name: String) = this should haveHeader(name)
+fun HttpResponse.shouldHaveHeader(name: String, value: String) = this should haveHeader(name, value)
+infix fun HttpResponse.shouldNotHaveHeader(name: String) = this shouldNot haveHeader(name)
+fun HttpResponse.shouldNotHaveHeader(name: String, value: String) = this shouldNot haveHeader(name, value)
 
 infix fun HttpResponse.shouldHaveVersion(version: HttpProtocolVersion) = this should haveVersion(version)
 infix fun HttpResponse.shouldNotHaveVersion(version: HttpProtocolVersion) = this shouldNot haveVersion(version)
-fun haveVersion(version: HttpProtocolVersion) = object : Matcher<HttpResponse> {
+private fun haveVersion(version: HttpProtocolVersion) = object : Matcher<HttpResponse> {
```

Comment:
We can't make these private. There might be users that use the matchers directly